### PR TITLE
Fix CUDA and DDP training

### PR DESCRIPTION
**Pull request by majercakdavid**
## Purpose
Update package used for cuda runtime for PyTorch according to installation instructions. Provide fixes for DDP training for some samples

## What is the expected review turnaround time?

Urgency:
- [X] High (needs review today)
- [ ] Medium (needs review within a few days - most common)
- [ ] Low (can wait a week)

## Does this introduce a breaking change?

```
[X] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: examples/components/CCFRAUD/traininsilo/run.py [hunk 1]
```diff
@@ -598,7 +598,7 @@ def run(args):
     )
     trainer.execute(args.checkpoint)
 
-    if torch.cuda.is_available() or int(os.environ["WORLD_SIZE"]) > 1:
```
Comment (garg-amit) [inline examples/components/CCFRAUD/traininsilo/run.py:601]: let's handle a case if there's no `WORLD_SIZE` env var and apply the changes across all examples.

Comment (jfomhover) [reply examples/components/CCFRAUD/traininsilo/run.py:601]: yep could we use os.environ.get("WORLD_SIZE", 1) so it would work even if the env var is not set?

Comment (jfomhover) [reply examples/components/CCFRAUD/traininsilo/run.py:601]: (I actually don't remember if the variable is set when the world size is actually just 1 and the job is not distributed)

File: examples/components/NER/traininsilo/run.py [hunk 2]
```diff
@@ -180,13 +180,13 @@ def __init__(
         logger.info(f"Trainable parameters: {trainable_params}")
 
         self.model_.train()
+        self.model_.to(self.device_)
```
Comment (jfomhover) [inline examples/components/NER/traininsilo/run.py:183]: Is .train() interfering with .to() somehow? I wonder if it's more proper to first do .to() then only .train()

File: examples/components/NER/traininsilo/run.py [hunk 3]
```diff
@@ -180,13 +180,13 @@ def __init__(
...
         if self._distributed:
             self.model_ = DDP(
```
Comment (jfomhover) [inline examples/components/NER/traininsilo/run.py:185]: You're moving the call to .to(device) much earlier than before, is there a reason for it?

In particular, when you instanciate DDP(), I don't know how that works with .to(), is it being taken care of internally in DDP()?

Comment (jfomhover) [reply examples/components/NER/traininsilo/run.py:185]: forget about this comment, no pb here

File: examples/components/NER/traininsilo/run.py (hunk 2 above)
Comment (majercakdavid) [reply examples/components/NER/traininsilo/run.py:183]: Should not matter according to documentation: https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module.train 

The only reason why this was moved here from line 189 is that DDP is conditioned under having model in GPU.

Comment (garg-amit) [reply examples/components/NER/traininsilo/run.py:182]: If this is not required for DDP, we can move this to line 194 as a model has to be in the `training mode` when applying changes for `differential-privacy`.(By default, huggingface loads a model in the `eval mode`)

**Review by garg-amit — APPROVED** [review]

**Review by garg-amit — APPROVED** [review]
let's update arm templates in a separate PR

**Review by jfomhover — APPROVED** [review]
